src/core.jl

```diff
@@ -4,6 +4,14 @@ accum_type(::Type{T}) where {T<:Real}       = Float64
 accum_type(::Type{T}) where {T<:FixedPoint} = floattype(T)
 accum_type(::Type{C}) where {C<:Colorant}   = base_colorant_type(C){accum_type(eltype(C))}
 
+accum_type(val) = isa(val, Type) ? throw_accum_type(val) : convert(accum_type(typeof(val)), val)
+throw_accum_type(T) = error("type $T not supported in `accum_type`")
+
+_abs2(c::MathTypes) = c ⋅ c
```

Comment:
Maybe stick to the old usage for now?
```suggestion
_abs2(c::MathTypes) = mapreducec(v->float(v)^2, +, 0, c)
```

Reply:
That sounds like a good idea.

Reply:
That seems to fix things, by and large. The few differences:
- the horse demo with `felzenszwalb(img, 100)` merges grass and sky, resulting in a dull gray used for both. That's true here (with your suggested change) *and* on the released version of the package. Both also give 43 segments, making me think that this probably changed some time ago but the image in the docs hasn't been updated. `felzenszwalb(img, 10)` produces the result shown in the image.
- There's now an off-by-one in the number of segments in https://juliaimages.org/stable/pkgs/segmentation/#Fast-Scanning (2539 instead of 2538, and 64 instead of 65). There is no visible change in the quality of the result, so to me that seems acceptable.
- New Julia releases change the order of random numbers for a given seed, and hence the coloration of images for the demos using `get_random_color`.